__tests__/server.test.js

```diff
@@ -191,67 +191,85 @@ describe("/api/articles", () => {
+  test("?order_by=ASC: Should return all articles objects with all properties sorted by default and ordered by asecending", () => {
+    return request(app)
+      .get("/api/articles?order_by=ASC")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { ascending: true });
+      });
+  });
+  test("?sort_by=votes&order_by=ASC: Should return all articles objects with all properties sorted by votes and ordered by asecending", () => {
+    return request(app)
+      .get("/api/articles?sort_by=votes&order_by=ASC")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("votes", { ascending: true });
+      });
+  });
+  test("?invalid sort_by 200: Should return all articles objects with all properties sorted by and ordered by default", () => {
+    return request(app)
+      .get("/api/articles?sort_by=not_a_valid_sort")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
+  test("?invalid order_by 200: Should return all articles objects with all properties sorted by and ordered by default", () => {
+    return request(app)
+      .get("/api/articles?order_by=not_a_valid-_order")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+      });
+  });
   test("?topic=mictch 200: Should return all articles with the topic of mitch with properties sorted and ordered by default", () => {
     return request(app)
-    .get("/api/articles?topic=mitch")
-    .expect(200)
-    .then(({body}) => {
-      expect(body.articles).toBeSortedBy("created_at", {descending: true,})
-      expect(body.articles.length).toBe(12)
-      body.articles.forEach((article) => {
-        expect(article.topic).toBe('mitch')
-      })
-    })
-  })
-  
+      .get("/api/articles?topic=mitch")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles).toBeSortedBy("created_at", { descending: true });
+        expect(body.articles.length).toBe(12);
+        body.articles.forEach((article) => {
+          expect(article.topic).toBe("mitch");
+        });
+      });
+  });
+  test("?topic=paper 200: Should return an empty array of articles whern passed a topic with no articles associated", () => {
+    return request(app)
+      .get("/api/articles?topic=paper")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(0);
+      });
+  });
+  test("?topic=not_a_topic 200: Should return all articles if passed an invlaid topic as it will be accepted as not passing a topic", () => {
+    return request(app)
+      .get("/api/articles?topic=not_a_topic")
+      .expect(200)
+      .then(({ body }) => {
+        expect(body.articles.length).toBe(13);
+        expect(body.articles).toBeSortedBy("created_at", {
+          descending: true,
+        });
+      });
+  });
```

Comment:
If the topic does not exist in the database the API should return an error of `404` _not found_ so that the user is being informed that the request topic does not exist